<h1>Palmer Penguins Data Analytics</h1>
<hr>
<h2>Tanay, Xynex Kim R.</h2>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3>Importing dataset to Jupyter Notebook</h3>
<br>
<h3>Conecting dataset to dataframe</h3>
<li>Name of the dataframe</li>
<li>Source</li>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h2>Descriptive Analytics</h2>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [6]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


<h3>Cleaning the Data frame</h3>

In [7]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [8]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Selecting a Column</h3>
<p>Syntax</p>

<P>data_frame['column_name]</P>

In [9]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [10]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [11]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [12]:
penguins[penguins['species'] == "Chinstrap"]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [13]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [14]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [15]:
chinstrap_dataf = penguins[penguins['species'] == "Chinstrap"]

In [16]:
chinstrap_dataf

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [17]:
adelie_dataf = penguins[penguins['species'] == "Adelie"]

In [18]:
adelie_dataf

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [19]:
gentoo_dataf = penguins[penguins['species'] == "Gentoo"]

In [20]:
gentoo_dataf

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [21]:
(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [22]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [23]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [24]:
chinstrap_heavy = penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

In [25]:
chinstrap_heavy

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

<h3>How to Fill up null values</h3>

In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [27]:
penguins['bill_length_mm']

0      39.1
1      39.5
2      40.3
3       NaN
4      36.7
       ... 
339    55.8
340    43.5
341    49.6
342    50.8
343    50.2
Name: bill_length_mm, Length: 344, dtype: float64

In [28]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [29]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [30]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [31]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [32]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

<h3>How to change Data type of a Column</h3>

In [33]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [34]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [35]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h3>How to create a Column</h3>

In [36]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [37]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [38]:
penguins['mass_kg'] = penguins['body_mass_g'] / 1000

In [39]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year   mass_kg  
0    3750.000000    male  2007  3.750000  
1    3800.000000  female  2007  3.800000  
2    3250.000000  female  2007  3.250000  
3    4201.754386    male  2007  4.201754  
4    3450.000000  female  2007  3.450000  
..           ...     ...   ...       ...  
339  4000.000000    male  2009  4.000000  
340  3400.000000  female  2009  3.400000  
341  3775.000000    male  2009  3.775000  
342  4100.000000    male  2009  4.100000  
343  3775.000000  female  2009  3.775000  

[344 rows x 9 columns]

In [40]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [41]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h3>Exploratory Data Analytics</h3>
<br>
<h3>Ex. 1. Which species has the heaviest body mass in kg?</h3>

In [42]:
chinstrap_dataf['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [43]:
adelie_dataf['body_mass_g'].mean()

np.float64(3700.662251655629)

In [44]:
gentoo_dataf['body_mass_g'].mean()

np.float64(5076.016260162602)

<h3>Insight #1</h3>

<h3>The heaviest species from the dataset is the GENTOO specie.</h3>

<h3>01. How many females are there in the Chinstrap species?</h3>

In [45]:
chinstrap_dataf['sex'].value_counts()

sex
female    34
male      34
Name: count, dtype: int64

<h3>Insight: The dataset shows that Chinstrap species has 34 females.</h3>

<h3>02. Which island has the highest population of species?</h3>

In [46]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>Insight: The Adelie penguins has the largest population among the three.</h3>

<h3>03. Among Chinstrap penguins are there many individual with a bill lenght of 40 mm?</h3>

In [47]:
chinstrap_bill = penguins[(penguins['species'] == "Chinstrap") & (penguins['bill_length_mm'] >= 40)]

In [48]:
chinstrap_bill

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
276       3500.0  female  2007      3.5  
277       3900.0    male  2007      3.9  
278       3650.0    male  2007      3.6  
279       3525.0  female  2007      3.5  
280       3725.0    male  2007      3.7  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[68 rows x 9 columns]

<h3>Insight: There are 68 Chinstrap penguins with a bill length greater than 40 mm.</h3>

<h3>04. Which species has the largest number of males?</h3>

In [49]:
chinstrap_dataf['sex'].value_counts()

sex
female    34
male      34
Name: count, dtype: int64

In [50]:
adelie_dataf['sex'].value_counts()

sex
male      73
female    73
Name: count, dtype: int64

In [51]:
gentoo_dataf['sex'].value_counts()

sex
male      61
female    58
Name: count, dtype: int64

<h3>Insight: The Adelie penguins have the highest number of males.</h3>

<h3>05.Which year have the highest data collected in Chinstrap?</h3>

In [52]:
chinstrap_dataf['year'].value_counts()

year
2007    26
2009    24
2008    18
Name: count, dtype: int64

<h3>Insight: The year 2007 have the highest recorded data which has 26 records.</h3>

<h3>06. Which species have the longest bill length?</h3>

In [53]:
chinstrap_dataf['bill_length_mm'].mean()

np.float64(48.83382352941177)

In [54]:
adelie_dataf['bill_length_mm'].mean()

np.float64(38.79139072847682)

In [55]:
gentoo_dataf['bill_length_mm'].mean()

np.float64(47.50487804878048)

<h3>Insight: Chinstrap have the longest bill length among the three.</h3>

<h3>07. Which year recorded have the highest number of species?</h3>

In [56]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h3>Insight: Down there shows the all in all numbers of recorded penguin species per year, also show that in the year of 2009 have the highest number of record that had collected.</h3>
<br>
<p>2007: 110</p>
<p>2008: 114</p>
<p>2009: 120</p>

In [57]:
penguins.describe(include='all')

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      344.000000     344.000000         344.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN   
mean       NaN     NaN       43.921930      17.151170         200.915205   
std        NaN     NaN        5.443643       1.969027          14.020657   
min        NaN     NaN       32.100000      13.100000         172.000000   
25%        NaN     NaN       39.275000      15.600000         190.000000   
50%        NaN     NaN       44.250000      17.300000         197.000000   
75%        NaN     NaN       48.500000      18.700000         213.000000   
max        NaN     NaN       59.600000      21.500000         231.000000   

        body_mass_g   sex    year     mass_kg  
count    344.000000   344   344.0  344.000000  
unique          NaN     2     3.0         NaN  
top             NaN  male  2009.0         NaN  
freq            NaN   179   120.0         NaN  
mean    4201.754386   NaN     NaN    4.202035  
std      799.613058   NaN     NaN    0.800744  
min     2700.000000   NaN     NaN    2.700000  
25%     3550.000000   NaN     NaN    3.600000  
50%     4050.000000   NaN     NaN    4.000000  
75%     4750.000000   NaN     NaN    4.800000  
max     6300.000000   NaN     NaN    6.300000

In [58]:
penguins.head()

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen        39.10000       18.70000         181.000000   
1  Adelie  Torgersen        39.50000       17.40000         186.000000   
2  Adelie  Torgersen        40.30000       18.00000         195.000000   
3  Adelie  Torgersen        43.92193       17.15117         200.915205   
4  Adelie  Torgersen        36.70000       19.30000         193.000000   

   body_mass_g     sex  year  mass_kg  
0  3750.000000    male  2007      3.8  
1  3800.000000  female  2007      3.8  
2  3250.000000  female  2007      3.2  
3  4201.754386    male  2007      4.2  
4  3450.000000  female  2007      3.4

In [59]:
penguins.head(20)

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0   Adelie  Torgersen        39.10000       18.70000         181.000000   
1   Adelie  Torgersen        39.50000       17.40000         186.000000   
2   Adelie  Torgersen        40.30000       18.00000         195.000000   
3   Adelie  Torgersen        43.92193       17.15117         200.915205   
4   Adelie  Torgersen        36.70000       19.30000         193.000000   
5   Adelie  Torgersen        39.30000       20.60000         190.000000   
6   Adelie  Torgersen        38.90000       17.80000         181.000000   
7   Adelie  Torgersen        39.20000       19.60000         195.000000   
8   Adelie  Torgersen        34.10000       18.10000         193.000000   
9   Adelie  Torgersen        42.00000       20.20000         190.000000   
10  Adelie  Torgersen        37.80000       17.10000         186.000000   
11  Adelie  Torgersen        37.80000       17.30000         180.000000   
12  Adelie  Torgersen        41.10000       17.60000         182.000000   
13  Adelie  Torgersen        38.60000       21.20000         191.000000   
14  Adelie  Torgersen        34.60000       21.10000         198.000000   
15  Adelie  Torgersen        36.60000       17.80000         185.000000   
16  Adelie  Torgersen        38.70000       19.00000         195.000000   
17  Adelie  Torgersen        42.50000       20.70000         197.000000   
18  Adelie  Torgersen        34.40000       18.40000         184.000000   
19  Adelie  Torgersen        46.00000       21.50000         194.000000   

    body_mass_g     sex  year  mass_kg  
0   3750.000000    male  2007      3.8  
1   3800.000000  female  2007      3.8  
2   3250.000000  female  2007      3.2  
3   4201.754386    male  2007      4.2  
4   3450.000000  female  2007      3.4  
5   3650.000000    male  2007      3.6  
6   3625.000000  female  2007      3.6  
7   4675.000000    male  2007      4.7  
8   3475.000000    male  2007      3.5  
9   4250.000000    male  2007      4.2  
10  3300.000000    male  2007      3.3  
11  3700.000000    male  2007      3.7  
12  3200.000000  female  2007      3.2  
13  3800.000000    male  2007      3.8  
14  4400.000000    male  2007      4.4  
15  3700.000000  female  2007      3.7  
16  3450.000000  female  2007      3.4  
17  4500.000000    male  2007      4.5  
18  3325.000000  female  2007      3.3  
19  4200.000000    male  2007      4.2

<h3>Data Aggregation</h3>

<h3>The use of .groupby()</h3>

In [60]:
PerIsland = penguins.groupby(['species','island']).count()

In [61]:
PerIsland

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Biscoe               0    0     0        0  
          Dream               68   68    68       68  
          Torgersen            0    0     0        0  
Gentoo    Biscoe             124  124   124      124  
          Dream                0    0     0        0  
          Torgersen            0    0     0        0

In [62]:
PerIsland = penguins.groupby(['species'])['island'].count()

In [63]:
PerIsland

species
Adelie       152
Chinstrap     68
Gentoo       124
Name: island, dtype: int64

In [64]:
penguins_insight8 = penguins.groupby(['species','island','sex','year']).size()

In [65]:
penguins_insight8

species    island     sex     year
Adelie     Biscoe     female  2007     5
                              2008     9
                              2009     8
                      male    2007     5
                              2008     9
                              2009     8
           Dream      female  2007     9
                              2008     8
                              2009    10
                      male    2007    11
                              2008     8
                              2009    10
           Torgersen  female  2007     8
                              2008     8
                              2009     8
                      male    2007    12
                              2008     8
                              2009     8
Chinstrap  Biscoe     female  2007     0
                              2008     0
                              2009     0
                      male    2007     0
                              2008     0
                      

In [66]:
avg_species_island = penguins.groupby(['species','island'])['body_mass_g'].mean()

In [67]:
avg_species_island

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3715.899123
Chinstrap  Biscoe               NaN
           Dream        3733.088235
           Torgersen            NaN
Gentoo     Biscoe       5068.965761
           Dream                NaN
           Torgersen            NaN
Name: body_mass_g, dtype: float64

<h2>Testing Correlation</h2>

<h3>Pearson R Correlation</h3>
<hr>
<p>Used for testing correlation between two<strong> NUMERICAL VALUES</strong></p>

In [68]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [69]:
weight_flipper_cor

np.float64(0.8712017673060114)

In [70]:
penguins.dtypes

species              category
island               category
bill_length_mm        float64
bill_depth_mm         float64
flipper_length_mm     float64
body_mass_g           float64
sex                  category
year                 category
mass_kg               float64
dtype: object

<h2>Numerical - Categorical Correlation Test</h2>
<h3>Library - SciPy (Scientific Python)</h3>
<hr>
<P>from scipy.stats import f_oneway</P>

In [71]:
from scipy.stats import f_oneway

In [72]:
penguins[penguins['sex'] == 'male']

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
5       Adelie  Torgersen        39.30000       20.60000         190.000000   
7       Adelie  Torgersen        39.20000       19.60000         195.000000   
8       Adelie  Torgersen        34.10000       18.10000         193.000000   
..         ...        ...             ...            ...                ...   
334  Chinstrap      Dream        50.20000       18.80000         202.000000   
336  Chinstrap      Dream        51.90000       19.50000         206.000000   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   

     body_mass_g   sex  year  mass_kg  
0    3750.000000  male  2007      3.8  
3    4201.754386  male  2007      4.2  
5    3650.000000  male  2007      3.6  
7    4675.000000  male  2007      4.7  
8    3475.000000  male  2007      3.5  
..           ...   ...   ...      ...  
334  3800.000000  male  2009      3.8  
336  3950.000000  male  2009      4.0  
339  4000.000000  male  2009      4.0  
341  3775.000000  male  2009      3.8  
342  4100.000000  male  2009      4.1  

[179 rows x 9 columns]

In [73]:
x_df = penguins[penguins['sex'] == 'male']

In [74]:
x_df['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [75]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [76]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
 8   mass_kg            344 non-null    float64 
dtypes: category(4), float64(5)
memory usage: 15.4 KB


<h3>08. Is the a correlation between penguins bill length and bill depth?</h3>

In [77]:
bill_cor = penguins['bill_depth_mm'].corr(penguins['bill_length_mm'])

In [78]:
bill_cor

np.float64(-0.23505287035553285)

<h3>Insight: The data above shows that <strong>there's no</strong> correlation between the bill length and depth.</h3>

<h3>09. Is there a correlation between the species sex and flipper length?</h3>

In [79]:
penguins[penguins['sex'] == 'male']['flipper_length_mm'].sum()

np.float64(36549.83040935673)

In [80]:
penguins[penguins['sex'] == 'female']['flipper_length_mm'].sum()

np.float64(32565.0)

In [83]:
sex_flipperL = f_oneway(penguins[penguins['sex'] == 'male']['flipper_length_mm'],penguins[penguins['sex'] == 'female']['flipper_length_mm'])

In [86]:
sex_flipperL

F_onewayResult(statistic=np.float64(21.5666589289189), pvalue=np.float64(4.88065474065626e-06))

<h3>Insight: <strong>There is</strong> a correlation in flipper length to both sexes; however, if we focus on which sex shows a stronger correlation with flipper length, the data indicates that <strong>males have the higher correlation</strong>.</h3>

<h3>10. Is there a correlation between the mass kg that Chinstrap and Gentoo has?</h3>

In [87]:
f_oneway(penguins[penguins['species'] == 'Gentoo']['mass_kg'],penguins[penguins['species'] == 'Chinstrap']['mass_kg'])

F_onewayResult(statistic=np.float64(356.0159739191799), pvalue=np.float64(1.9986868807711357e-45))

<h3>Insight: There is a correlation between Chinstrap and Gentoo penguins in terms of body mass (kg).</h3>

<h3>11. Would there be a correlation between all the species in terms of bill depth?</h3>

In [93]:
f_oneway(penguins[penguins['species'] == 'Adelie']['bill_depth_mm'],penguins[penguins['species'] == 'Chinstrap']['bill_depth_mm'])

F_onewayResult(statistic=np.float64(0.22278628331837413), pvalue=np.float64(0.6373971676087197))

In [94]:
f_oneway(penguins[penguins['species'] == 'Gentoo']['bill_depth_mm'],penguins[penguins['species'] == 'Chinstrap']['bill_depth_mm'])

F_onewayResult(statistic=np.float64(468.361544482491), pvalue=np.float64(3.651652135731056e-53))

In [95]:
f_oneway(penguins[penguins['species'] == 'Gento']['bill_depth_mm'],penguins[penguins['species'] == 'Adelie']['bill_depth_mm'])

F_onewayResult(statistic=np.float64(nan), pvalue=np.float64(nan))

<h3>Insight: The data indicates a strong correlation among all species in terms of bill depth</h3>

<h3>12. Is there a correlation between the Adelie and Chinstrap species in terms of bill length (mm)?</h3>

In [96]:
f_oneway(penguins[penguins['species'] == 'Adelie']['bill_length_mm'],penguins[penguins['species'] == 'Chinstrap']['bill_length_mm'])

F_onewayResult(statistic=np.float64(558.4183971664146), pvalue=np.float64(4.724199491448672e-62))

<h3>Insight: Chinstrap and Adelie penguins show a clear correlation in bill length</h3>